### DISCUSSION TITLE
move to Yarn Modern

### DISCUSSION THREAD

**Discussion by fembau** [research-software-directory/RSD-as-a-service#843] [Ideas]
I was wondering why I had untracked files after running `make install`...

:bulb: On my fresh installed system I have a recent version of yarn (v3.5), which for example creates a `.yarn` folder, while the project is on v1.

https://yarnpkg.com/getting-started/qa

**Comment by dmijatovic** [comment]
Good point. I am not sure about yarn. It became popular as v1 when npm was having some performance issues. Later npm did improved it's performance and the difference became smaller. 

I prefer (plain) npm. Beside yarn another alternative could be pnpm. I am not sure how much time is needed to learn any of these alternatives. I use these tools just to install dependencies. npm is required at minimum, the other solutions go on the top of node/npm asfaik. 

I would prefer keeping it to the basics in the future (npm) but I am also interested to hear what others think.

Proper cost-benefit analyses is desired.

**Comment by fembau** [comment]
As far as I remember you can not use npm and yarn simultaneously because both have their own package lock file – which is also commited. Thus it could happen that the npm user installs other package versions than the yarn user or the other way round. Only users which use the package manager of the commited lock file will have the same package versions installed.

**Reply by dmijatovic** [reply]
> As far as I remember you can not use npm and yarn simultaneously because both have their own package lock file – which is also commited. Thus it could happen that the npm user installs other package versions than the yarn user or the other way round. Only users which use the package manager of the commited lock file will have the same package versions installed.

Yes. That is correct. We actually started with npm and then some team members were using yarn v1. To prevent having 2 lockfiles we decided to use yarn. But if we should change something my preference would be to move back to npm. However, I think most suitable for us would be to use package manager that is most widely used. With that in mind we could investigate current usage and popularity of package managers (as I have not done this lately).

**Comment by dmijatovic** [comment]
In the near future I will be looking at the splitting (monolith) front end into a number of modules. The primary goal is related to enable other hosts to have the completely custom homepage. Indirectly this request requires making of separate modules. For the maximum flexibility I think about splitting ui into following top level modules: home, software, projects, organisations, user and admin. 

In order to manage these module I tend to gravitate toward monorepo workspaces and PNPM in order to manage node_module folders of all these applications. I will come with the proposal but **concerning YARN, NPM and PNPM** discussion I will be advocating to **switch to PNPM (including workspaces approach)** in the future.

@jmaassen @fembau @ewan-escience @ctwhome @cmeessen What do you think? Do you have any **strong** preference between NPM/YARN/PNPM?

**Comment by dmijatovic** [comment]
Maybe use PNPM as the only dependency? It is possible then to use pnpm to install node. In general pnpm can replace npm, yarn and nvm. See this article https://pawelgrzybek.com/i-replaced-npm-yarn-and-nvm-with-pnpm/
